sdk/appconfiguration/app-configuration/test/public/secretReference.spec.ts

```diff
@@ -165,7 +173,11 @@ describe("AppConfigurationClient - SecretReference", () => {
         0,
         "Unexpected number of SecretReferences seen after updating"
       );
-      await client.deleteConfigurationSetting({ key: secondSetting.key });
+      await client.setReadOnly({ key: secondSetting.key, label: secondSetting.label }, false);
```

Comment:
we may not need to do this extra request if we know the readOnly was false?

So, add an if check to check that?